Sources/Confidence/Confidence.swift

```diff
@@ -133,12 +140,23 @@ public class Confidence: ConfidenceEventSender {
     - Parameter defaultValue: returned in case of errors or in case of the variant's rule indicating to use the default value.
     */
     public func getEvaluation<T>(key: String, defaultValue: T) -> Evaluation<T> {
-        self.cache.evaluate(
-            flagName: key,
-            defaultValue: defaultValue,
-            context: getContext(),
-            flagApplier: flagApplier
-        )
+        cacheQueue.sync {  [weak self] in
+            guard let self = self else {
+                return Evaluation(
```

Comment:
Add some logging on warning level here perhaps?

Reply:
We don't have access to the logger if `self` is lost :(

Reply:
Ah, of course :(